### PR TITLE
NXP-26149: allow to configure a custom WOPI base URL

### PR DESCRIPTION
**Pull request by troger**

### CODE REVIEW

File: nuxeo-wopi/src/test/java/org/nuxeo/wopi/jaxrs/TestFilesEndpoint.java [hunk 1]
```diff
@@ -836,30 +839,49 @@ public void testPutRelativeFile() throws IOException {
...
+    protected void assertPutRelativeFileResponse(CloseableClientResponse response, String newName, String wopiBaseURL)
+            throws IOException {
         assertEquals(200, response.getStatus());
         JsonNode node = mapper.readTree(response.getEntityInputStream());
         assertEquals(newName, node.get(NAME).asText());
-        assertTrue(node.has(URL));
-        assertTrue(node.has(HOST_VIEW_URL));
-        assertTrue(node.has(HOST_EDIT_URL));
+        String url = node.get(URL).asText();
+        assertNotNull(url);
```
Comment (ataillefer) [inline nuxeo-wopi/src/test/java/org/nuxeo/wopi/jaxrs/TestFilesEndpoint.java]: Shouldn't the not null assertion be on `node.get(URL)`, then get the text?

File: nuxeo-wopi/src/test/java/org/nuxeo/wopi/jaxrs/TestFilesEndpoint.java [hunk 2]
```diff
@@ -836,30 +839,49 @@ public void testPutRelativeFile() throws IOException {
...
         try (CloseableClientResponse response = post(johnToken, data, headers, blobDocFileId)) {
-            assertPutRelativeFileResponse(response, "foo.docx");
+            assertPutRelativeFileResponse(response, "foo.docx", BASE_URL);
+        }
+
+        try {
+            String customWOPIBaseURL = "http://foo";
+            // success - 200 - new file from suggested filename with a custom WOPI base URL
+            Framework.getProperties().setProperty(WOPI_BASE_URL_PROPERTY, customWOPIBaseURL);
+            headers.put(SUGGESTED_TARGET, "foo.docx");
```
Comment (ataillefer) [inline nuxeo-wopi/src/test/java/org/nuxeo/wopi/jaxrs/TestFilesEndpoint.java]: Not needed as already set

**Comment by ataillefer** [comment]
:+1:

**Comment by kevinleturc** [comment]
👍

**Pull request merged**